ui/src/app/pages/workspace/workspace-wrapper.tsx

```diff
@@ -133,156 +132,154 @@ const MultiRegionWorkspaceNotification = () => {
-            maxCreateCount: 0,
-            maxResumeCount: 0,
-          });
-        } catch {
-          // Ignore InitialRuntimeNotFoundError.
-          // Ignore ExceededActionCountError. This is thrown when the runtime doesn't exist, or
-          // isn't started. Both of these scenarios are expected, since we don't want to do any lazy
-          // initialization here.
-          // Also ignore LeoRuntimeInitializationAbortedError - this is expected when navigating
-          // away from a page during a poll.
-          // Ideally, we would handle or log errors except the ones listed above.
-        }
-      };
-      const getWorkspaceAndUpdateStores = async (namespace, id) => {
-        try {
-          // No destructuring because otherwise it shadows the workspace in props
-          const wsResponse = await workspacesApi().getWorkspace(namespace, id);
-          currentWorkspaceStore.next({
-            ...wsResponse.workspace,
-            accessLevel: wsResponse.accessLevel,
-          });
-          updateStores(wsResponse.workspace.namespace);
-        } catch (ex) {
-          if (ex.status === 403) {
-            navigate(['/workspaces']);
-          }
+        updateStores(wsResponse.workspace.namespace);
+      } catch (ex) {
+        if (ex.status === 403) {
+          navigate(['/workspaces']);
         }
-      };
+      }
+    };
 
-      if (
-        !currentWorkspaceStore.getValue() ||
-        currentWorkspaceStore.getValue().namespace !== ns ||
-        currentWorkspaceStore.getValue().id !== wsid
-      ) {
-        currentWorkspaceStore.next(null);
-        // In a handful of situations - namely on workspace creation/clone,
-        // the application will preload the next workspace to avoid a redundant
-        // refetch here.
-        const nextWs = nextWorkspaceWarmupStore.getValue();
-        nextWorkspaceWarmupStore.next(undefined);
-        if (nextWs && nextWs.namespace === ns && nextWs.id === wsid) {
-          currentWorkspaceStore.next(nextWs);
-          updateStores(ns);
-        } else {
-          getWorkspaceAndUpdateStores(ns, wsid);
-        }
+    if (
+      !currentWorkspaceStore.getValue() ||
+      currentWorkspaceStore.getValue().namespace !== ns ||
+      currentWorkspaceStore.getValue().id !== wsid
+    ) {
+      currentWorkspaceStore.next(null);
+      // In a handful of situations - namely on workspace creation/clone,
+      // the application will preload the next workspace to avoid a redundant
+      // refetch here.
+      const nextWs = nextWorkspaceWarmupStore.getValue();
+      nextWorkspaceWarmupStore.next(undefined);
+      if (nextWs && nextWs.namespace === ns && nextWs.id === wsid) {
+        currentWorkspaceStore.next(nextWs);
+        updateStores(ns);
+      } else {
+        getWorkspaceAndUpdateStores(ns, wsid);
       }
-    }, [ns, wsid]);
+    }
+  }, [ns, wsid]);
+  const workspace = currentWorkspaceStore.getValue();
```

Comment:
I added this line here. Previously, workspace is from `withCurrentWorkspace`. I change to `currentWorkspaceStore.getValue()` and I think by this time currentWorkspaceStore already have correct workspace. Is that correct?

Reply:
I don't think we are guaranteed to have the `currentWorkspaceStore` correctly initialized at this point.

What I'd prefer is a state variable, like `const [workspace, setWorkspace] = useState<Workpace>(undefined)` which we can set when we know we have a valid workspace, like inside the `useEffect` somewhere